### DISCUSSION TITLE
Hashicorp vault running on AWS EC2 instance and trying to integrate the vault with Github actions using OIDC and JWT

### DISCUSSION THREAD

**Discussion by scharishma-db** [community/community#48985] [Actions]
### Select Topic Area

Product Feedback

### Body

ERROR message:
::endgroup::
Error: connect ECONNREFUSED 127.0.0.1:8200
##[debug]Node Action run completed with exit code 1
To Reproduce
The yaml of the vault-action step, with any sensitive information masked or removed.

    - name: Get Secrets
      id: secrets
      uses: hashicorp/vault-action@v2.4.0
      with:
        url: ${{ secrets.VAULT_ADDR }}
        token: ${{ secrets.VAULT_TOKEN }}
        caCertificate: ${{ secrets.VAULTCERT }}
        role: vault-github-poc-role
        method: jwt
        # path: gh-actions
        secrets: |
           secret/data/gh-vault password

kubernetesTokenPath: /var/run/secrets/kubernetes.io/serviceaccount/token
exportEnv: true
exportToken: false
tlsSkipVerify: false
jwtTtl: 3[60]
Get Vault Secrets
Error: connect ECONNREFUSED 127.0.0.1:8200
##[debug]Node Action run completed with exit code 1
##[debug]Finishing: Get Secrets

Additional context

Question is can we connect the hashicorp vault hosted on was ec2 instance or vault running as container in Kubernetes to GitHub secrets workflow? Or do we need to do this only via self hosted runners, please provide more details

**Comment by scharishma-db** [comment]
can someone help on this?

**Comment by jsoref** [comment]
>Error: connect ECONNREFUSED 127.0.0.1:8200

That doesn't seem like an address of a particularly remote address...

If you properly bridge your workflow to your EC2 instance, then, yes, it should work.

How to do that is left as an exercise (it seems like you're also dealing w/ a k8s cluster).

You might consider using a self-hosted runner instead where the self-hosted runner lives in EC2 or EKS, that might be easier to reason through.
